tests/angular-localForage.js

```diff
@@ -465,4 +477,94 @@ describe('Module: LocalForageModule', function() {
       }, done);
     });
   });
+
+  describe("createInstance", function () {
+      beforeEach(function () {
+          $localForage.createInstance({
+              name: 'DUPLICATE_INSTANCE_NAME'
+          });
+      });
+      it('should create a new instance', function () {
+          expect($localForage.createInstance({
+              name: 'TEST_INSTANCE'
+          })).toBeDefined();
+      });
+
+      it('should throw error if trying to create duplicate instance.',
+         function () {
+          expect($localForage.createInstance.bind($localForage, {
+              name: 'DUPLICATE_INSTANCE_NAME'
+          })).toThrowError(/already defined/);
+      });
+
+      it('should create instance with same name, different storeName',
+         function () {
+          expect($localForage.createInstance.bind($localForage, {
+              name: 'DUPLICATE_INSTANCE_NAME',
+              storeName: 'DIFFERENT_STORE_NAME'
+          })).not.toThrowError(/already defined/);
+      });
+  });
+
+  describe("instance", function () {
+      var $q, interval;
+      beforeEach(function () {
+          $localForage.createInstance({
+              name: 'TEST_INSTANCE_1'
+          });
+          $localForage.createInstance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_1'
+          });
+          $localForage.createInstance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_2'
+          });
+          inject(function (_$q_) {
+              $q = _$q_;
+          });
+          interval = triggerDigests();
+      });
+
+      afterEach(function () {
+          stopDigests(interval);
+      });
+
+      it('should get instance by name', function () {
+          expect($localForage.instance({
+              name: 'TEST_INSTANCE_1'
+          })).toBeDefined();
+      });
+
+      it('should throw exception if instance not exists', function () {
+          expect($localForage.instance.bind($localForage, {
+              name: 'NON_EXISTENT'
+          })).toThrowError();
+      });
+
+      it('should get instances with same name, different storeNames',
+         function (done) {
+          var instance1 = $localForage.instance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_1'
+          });
+          var instance2 = $localForage.instance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_2'
+          });
+          $q.all([
```

Comment:
I have already applied your suggestion and changed the test to not.toBe(instance).

my example was not for the $localForage, but for localforage itself.

in the localforage library, there is no `instance` function, just `createInstance` and my example just works fine (and two same instances are different references too.). I think in localforage module, `createInstance` name is a bit confusing, needs to be `instance` instead. because it also takes care of `createInstance`. in angular-localForage, the similar behavior can be produced with a function;
```
function instance(opts) {
    try {
        return $localForage.createInstance(opts);
    } catch(e) {
        return $localForage.instance(opts);
    }
}
```